crates/alerter/src/lib.rs

```diff
@@ -23,30 +22,105 @@ use {
 #[serde_as]
 #[derive(Debug, serde::Deserialize, Eq, PartialEq)]
 #[serde(rename_all = "camelCase")]
-struct Order {
+struct BaseOrder {
     kind: OrderKind,
     buy_token: H160,
     #[serde_as(as = "HexOrDecimalU256")]
     buy_amount: U256,
     sell_token: H160,
     #[serde_as(as = "HexOrDecimalU256")]
     sell_amount: U256,
-    #[serde_as(as = "HexOrDecimalU256")]
-    fee_amount: U256,
     uid: OrderUid,
-    status: OrderStatus,
-    creation_date: DateTime<Utc>,
     partially_fillable: bool,
     #[serde(flatten)]
     class: OrderClass,
 }
 
-impl Order {
+impl BaseOrder {
     fn is_liquidity_order(&self) -> bool {
         matches!(self.class, OrderClass::Liquidity)
     }
 }
 
+#[serde_as]
+#[derive(Debug, serde::Deserialize, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+struct ModelOrder {
+    #[serde(flatten)]
+    base: BaseOrder,
+    status: OrderStatus,
```

Comment:
Not sure if the differentiation between these 2 different order types is so significant.
If you had a single order type with `status: Option<OrderStatus>` you could get rid of a bunch of boiler plate logic.

Reply:
Exactly. I removed the remaining fields at the end so completely became ignorant that I was left with only one field of difference :)
Anyway, updated.